context/app/api/vitessce_confs/base_confs.py

```diff
@@ -428,3 +427,62 @@ def _setup_view_config_raster_cellsets_expression_segmentation(self, vc, dataset
             variablesLabelOverride="antigen", transpose=True
         )
         return vc
+
+
+class MultiImageSPRMAnndataViewConfigError(Exception):
+    """Raised when one of the individual SPRM view configs errors out"""
+    pass
+
+
+class MultiImageSPRMAnndataViewConfBuilder(ViewConfBuilder):
+    """Wrapper class for generating multiple "second generation" AnnData-backed SPRM
+    Vitessce configurations via SPRMAnnDataViewConfBuilder,
+    used for datasets with multiple regions.
+    """
+
+    def __init__(self, entity, groups_token, is_mock=False):
+        super().__init__(entity, groups_token, is_mock)
+        self._expression_id = 'expr'
+        self._mask_id = 'mask'
+        self._image_pyramid_subdir_regex = SPRM_PYRAMID_SUBDIR
+        self._mask_pyramid_subdir_regex = SPRM_PYRAMID_SUBDIR.replace(
+            self._expression_id, self._mask_id
+        )
+
+    def _find_ids(self):
+        """Search the image pyramid directory for all of the names of OME-TIFF files
+        to use as unique identifiers.
+        """
+        file_paths_found = [file["rel_path"] for file in self._entity["files"]]
+        full_pyramid_path = IMAGE_PYRAMID_DIR + "/" + self._image_pyramid_subdir_regex
+        pyramid_files = [file for file in file_paths_found if full_pyramid_path in file]
+        found_ids = [re.sub(r".ome.tif(f?)", "", Path(image_path).name).replace(
```

Comment:
either:
- `\.ome\.tiff?`... and probably anchor the end?
- or might be simpler just not to bother with regexes?:
```
Path(image_path).name.replace('.ome.tiff', '').replace('.ome.tif', '').replace('_' + self._expression_id, '')